### ISSUE TITLE
Why not to use hash index for _bucket space

### ISSUE THREAD

**Issue by Khatskevich** [tarantool/vshard#128]
Hash index is much faster for huge tables and may be more efficient for the `_bucket` storage

**Comment by Gerold103** [comment]
Tbh I am not sure anymore that we need this ticket. Regardless of how you store the buckets, accessing an index is orders of magnitude slower than accessing a Lua table. Even if the index is hash. And we already store bucket_ref objects in a Lua table. So a proper solution would be to ensure that the bucket_refs are created on first access to a bucket and are kept cached until the bucket moves out. The latter should be supported by work in scope of #173.

In other words, this ticket is not needed if #285 is done.
